me.id.webverify/webverifylib/src/main/java/me/id/webverifylib/IDmeWebVerify.java

```diff
@@ -204,34 +200,49 @@ private String ExtractAccessToken(String url)
      */
     private void GetWebProfile(String url)
     {
+        String serverResponse;
+        HttpURLConnection urlConnection;
+        URL urlRequest;
+        try {
+            urlRequest = new URL(url);
+            urlConnection = (HttpURLConnection) urlRequest.openConnection();
+
+            int responseCode = urlConnection.getResponseCode();
+
+            if(responseCode == HttpURLConnection.HTTP_OK){
+                serverResponse = readStream(urlConnection.getInputStream());
+                SendDataBack(serverResponse);
+            }
+        } catch (IOException e) {
+            SendErrorBack(e.getMessage());
+        }
+    }
 
-        try
-        {
-            HttpClient httpclient = new DefaultHttpClient();
-            HttpResponse response = httpclient.execute(new HttpGet(url));
-            StatusLine statusLine = response.getStatusLine();
-
-            if (statusLine.getStatusCode() == HttpStatus.SC_OK)
-            {
-                ByteArrayOutputStream out = new ByteArrayOutputStream();
-                response.getEntity().writeTo(out);
-                String responseString = out.toString();
-
-                out.close();
-                SendDataBack(responseString);
-
+    /**
+     * This converts the InputStream to a String
+     *
+     * @param inputStream from the Web Request
+     * @return the converted string
+     */
+    private String readStream(InputStream inputStream) throws IOException {
+        BufferedReader reader = null;
+        StringBuilder response = new StringBuilder();
+        try {
+            reader = new BufferedReader(new InputStreamReader(inputStream));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                response.append(line);
             }
-            else
-            {
-                //Closes the connection.
-                response.getEntity().getContent().close();
-                throw new IOException(statusLine.getReasonPhrase());
+        } finally {
+            if (reader != null) {
+                try {
+                    reader.close();
+                } catch (IOException ignored) {
+
```

Comment:
This is a lot harmful. Are you 100% sure of this? What about throwing it up, ignoring the warning. Do you know what is the concern of the warning?

At least I think you should state why ignoring it. What if this happens? It would pass silently and would be very hard to debug.